src/DevAdventCalendarCompetition/DevAdventCalendarCompetition/Extensions/StartupExtensions.cs

```diff
@@ -17,12 +18,45 @@
 using Microsoft.EntityFrameworkCore;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
 using Microsoft.OpenApi.Models;
 
 namespace DevAdventCalendarCompetition.Extensions
 {
     public static class StartupExtensions
     {
+        public static IServiceCollection AddAdventConfiguration(this IServiceCollection services, IConfiguration configuration)
+        {
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            services.Configure<AdventSettings>(a =>
+                {
+                    string defaultDateTimeFormat = "dd-MM-yyyy";
+                    var isAdventEndDate = configuration.GetSection("Competition:EndDate").Value;
+                    var isAdventStartDate = configuration.GetSection("Competition:StartDate").Value;
+                    var isValidStartDateTime = DateTimeOffset.TryParseExact(isAdventStartDate, defaultDateTimeFormat,
+                        CultureInfo.InvariantCulture,
+                        DateTimeStyles.None, out var startDate);
+                    var isValidEndDateTime = DateTimeOffset.TryParseExact(isAdventEndDate, defaultDateTimeFormat,
+                        CultureInfo.InvariantCulture,
+                        DateTimeStyles.None, out var endDate);
+
+                    if (!(isValidStartDateTime && isValidEndDateTime))
+                    {
+                        throw new InvalidOperationException(ExceptionsMessages.WrongFormatOfDate);
```

Comment:
nice to add to exception parameter with default date format

Reply:
Do you suggest changing WrongFormatOfDate and add info about default date format?

Reply:
Si ;) add parameter wchis is defaultFormatDate variable